ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/streaming/TopicStreamWriterTest.java

```diff
@@ -0,0 +1,131 @@
+package io.confluent.ksql.rest.server.resources.streaming;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.common.primitives.Ints;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import java.io.OutputStream;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalInt;
+import java.util.Queue;
+import java.util.stream.IntStream;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.utils.Bytes;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TopicStreamWriterTest {
+
+  @Mock public KafkaConsumer<String, Bytes> kafkaConsumer;
+  @Mock public SchemaRegistryClient schemaRegistry;
+
+  @Before
+  public void setup() {
+    final Iterator<ConsumerRecords<String, Bytes>> records = IntStream.iterate(0, i -> i + 1)
+        .mapToObj(i -> new ConsumerRecord<>(
+            "topic", 0, i, String.valueOf(i), new Bytes(Ints.toByteArray(i))))
```

Comment:
It might be easier to read the test if we just used the string representation of i in both the key and value (maybe with some big offset for the key so we can validate the value is actually the value)